extensions/amp-moviads-player/0.1/amp-moviads-player.js

```diff
@@ -0,0 +1,890 @@
+    div.id = 'amp-ima-player-recommended';
+    if (this.config_.videoWall !== null) {
+      this.config_.videoWall.forEach(function(entry) {
+        const divContainer = document.createElement('div');
+        const ahref = document.createElement('a');
+        ahref.href = entry.uri;
+        const img = document.createElement('IMG');
+        img.src = entry.poster;
+        img.setAttribute('data-src', entry.uri);
+        ahref.appendChild(img);
+        const divTxt = document.createElement('div');
+        divTxt.className = 'title';
+        const ahrefDiv = document.createElement('a');
+        ahrefDiv.href = entry.uri;
+        ahrefDiv.appendChild(document.createTextNode(entry.title));
+        divTxt.appendChild(ahrefDiv);
+        divContainer.appendChild(ahref);
+        divContainer.appendChild(divTxt);
+        div.appendChild(divContainer);
+      });
+      this.element.appendChild(div);
+    }
+  }
+
+  /** @override */
+  initGA(ua, name) {
+    (function(i, s, o, g, r, a, m) {
+      i['GoogleAnalyticsObject'] = r;
+      (i[r] =
+        i[r] ||
+        function() {
+          (i[r].q = i[r].q || []).push(arguments);
+        }),
+        (i[r].l = Number(new Date()));
+      (a = s.createElement(o)), (m = s.getElementsByTagName(o)[0]);
+      a.async = 1;
+      a.src = g;
+      m.parentNode.insertBefore(a, m);
+    })(
+      window,
+      document,
+      'script',
+      'https://www.google-analytics.com/analytics.js',
+      'ga'
+    );
+    ga('create', ua, 'auto', name);
+    ga(name + '.send', 'pageview');
+  }
+
+  /** @override */
+  eventGA(event, name, UA) {
+    if (typeof ga !== 'undefined') {
+      if (UA && event !== 'pause') {
+        // eslint-disable-next-line no-undef
+        ga(
+          name + '.send',
+          'event',
+          'Videos',
+          event,
+          this.config_.movieContenerId,
+          {
+            nonInteraction: true,
+          }
+        );
+      }
+    }
+  }
+
+  /** @override */
+  functionAfterEnded() {
+    if (
+      this.config_.hideWhenEnd === false &&
+      typeof this.config_.functionAfterEnded !== 'undefined' &&
+      this.config_.functionAfterEnded !== null &&
+      this.config_.functionAfterEnded.length > 0 &&
+      typeof window[this.config_.functionAfterEnded] === 'function'
+    ) {
+      window[this.config_.functionAfterEnded]();
+    }
+  }
+
+  /** @override */
+  detectedAdBlock() {
+    const _this = this;
+    const testURLAdBlock = 'https://imasdk.googleapis.com/js/sdkloader/ima3.js';
+    const myInitHead = {
+      method: 'HEAD',
+      mode: 'no-cors',
+    };
+    const myRequest = new Request(testURLAdBlock, myInitHead);
+    fetch(myRequest)
+      .then(function() {
+        _this.adblockDetected_ = false;
+      })
+      .then(function() {
+        _this.adblockDetected_ = false;
+      })
+      .catch(function() {
+        _this.adblockDetected_ = true;
+        _this.infoAdblock();
+      });
```

Comment:
Please do this inside the frame, don't implement on AMP layer.

Reply:
I changed the date to 2021

Reply:
removed method